src/main.zig

```diff
@@ -154,6 +149,51 @@ pub fn mainArgs(gpa: *Allocator, arena: *Allocator, args: []const []const u8) !v
         }
     }
 
+    const is_cgo = std.os.getenvZ("ZIG_CGO") != null;
+    if (is_cgo) cgo: {
+        // Workaround for https://github.com/golang/go/issues/43078, here we fix the order of
+        // command line arguments. Dear Go developers please let the Zig project know when
+        // we can remove this!
```

Comment:
https://go-review.googlesource.com/c/go/+/276412/ which is under review may fix this.